exchanges/order/order_test.go

```diff
@@ -1496,109 +1492,55 @@ func TestValidationOnOrderTypes(t *testing.T) {
-		15: {Exchange: "Binance", Type: Limit, Status: New},
-		16: {Exchange: "Binance", Type: AnyType},
-		17: {AccountID: "8888"},
-	}
+	id := uuid.Must(uuid.NewV4())
 
-	orders := map[int]Detail{
-		0:  {},
-		1:  {Exchange: "Binance"},
-		2:  {InternalOrderID: id},
-		3:  {OrderID: "2222"},
-		4:  {ClientOrderID: "3333"},
-		5:  {ClientID: "4444"},
-		6:  {WalletAddress: "5555"},
-		7:  {Type: AnyType},
-		8:  {Type: Limit},
-		9:  {Side: AnySide},
-		10: {Side: Sell},
-		11: {Status: AnyStatus},
-		12: {Status: New},
-		13: {AssetType: asset.Spot},
-		14: {Pair: currency.NewPair(currency.BTC, currency.USD)},
-		15: {Exchange: "Binance", Type: Limit, Status: New},
-		16: {AccountID: "8888"},
-	}
-	// empty filter tests
-	emptyFilter := filters[0]
-	for _, o := range orders {
-		if !o.MatchFilter(emptyFilter) {
-			t.Error("empty filter should match everything")
-		}
-	}
+	assert.True(t, new(Detail).MatchFilter(&Filter{}), "an empty filter should match an empty order")
+	assert.True(t, (&Detail{Exchange: "E", OrderID: "A", Side: Sell, Pair: currency.NewBTCUSD()}).MatchFilter(&Filter{}), "an empty filter should match any order")
 
-	tests := map[int]struct {
-		f              *Filter
-		o              Detail
-		expectedResult bool
+	tests := []struct {
+		m string
+		f Filter
+		o Detail
+		e bool
 	}{
-		0:  {filters[1], orders[1], true},
-		1:  {filters[1], orders[0], false},
-		2:  {filters[2], orders[2], true},
-		3:  {filters[2], orders[3], false},
-		4:  {filters[3], orders[3], true},
-		5:  {filters[3], orders[4], false},
-		6:  {filters[4], orders[4], true},
-		7:  {filters[4], orders[5], false},
-		8:  {filters[5], orders[5], true},
-		9:  {filters[5], orders[6], false},
-		10: {filters[6], orders[6], true},
-		11: {filters[6], orders[7], false},
-		12: {filters[7], orders[7], true},
-		13: {filters[7], orders[8], true},
-		14: {filters[7], orders[9], true},
-		15: {filters[8], orders[7], false},
-		16: {filters[8], orders[8], true},
-		17: {filters[8], orders[9], false},
-		18: {filters[9], orders[9], true},
-		19: {filters[9], orders[10], true},
-		20: {filters[9], orders[11], true},
-		21: {filters[10], orders[10], true},
-		22: {filters[10], orders[11], false},
-		23: {filters[10], orders[9], false},
-		24: {filters[11], orders[11], true},
-		25: {filters[11], orders[12], true},
-		26: {filters[11], orders[10], true},
-		27: {filters[12], orders[12], true},
-		28: {filters[12], orders[13], false},
-		29: {filters[12], orders[11], false},
-		30: {filters[13], orders[13], true},
-		31: {filters[13], orders[12], false},
-		32: {filters[14], orders[14], true},
-		33: {filters[14], orders[13], false},
-		34: {filters[15], orders[15], true},
-		35: {filters[16], orders[15], true},
-		36: {filters[17], orders[16], true},
-		37: {filters[17], orders[15], false},
-	}
-	// specific tests
-	for num, tt := range tests {
-		t.Run(strconv.Itoa(num), func(t *testing.T) {
+		{"Exchange ✓", Filter{Exchange: "A"}, Detail{Exchange: "A"}, true},
```

Comment:
In the function we have an EqualFold call for exchange name matching, wondering if that is necessary.

Reply:
I'm always primed to remove Equalfold where it's not necessary.
Nothing about Match says it should work case insensitively.
GC and thrasher will get a chance to object if they disagree.

Fixed https://github.com/gbjk/gocryptotrader/commit/95665011d4